packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Banner/banner.js

```diff
@@ -0,0 +1,309 @@
+        wrapperStyles.backgroundSize = backgroundSize;
+        wrapperStyles.backgroundPosition = backgroundPosition;
+        wrapperStyles.backgroundAttachment = backgroundAttachment;
+        wrapperStyles.backgroundRepeat = backgroundRepeat
+            ? 'repeat'
+            : 'no-repeat';
+    }
+
+    if (appearance === 'poster') {
+        overlayStyles.borderRadius = borderRadius;
+        overlayStyles.minHeight = minHeight;
+        overlayStyles.paddingTop = paddingTop;
+        overlayStyles.paddingRight = paddingRight;
+        overlayStyles.paddingBottom = paddingBottom;
+        overlayStyles.paddingLeft = paddingLeft;
+        contentStyles.width = '100%';
+    } else {
+        wrapperStyles.minHeight = minHeight;
+        wrapperStyles.paddingTop = paddingTop;
+        wrapperStyles.paddingRight = paddingRight;
+        wrapperStyles.paddingBottom = paddingBottom;
+        wrapperStyles.paddingLeft = paddingLeft;
+    }
+
+    const appearanceClasses = {
+        poster: classes.poster,
+        'collage-left': classes.collageLeft,
+        'collage-centered': classes.collageCentered,
+        'collage-right': classes.collageRight
+    };
+
+    const typeToPriorityMapping = {
+        primary: 'high',
+        secondary: 'normal',
+        link: 'low'
+    };
+
+    let linkProps = {};
+    let url = '';
+    let LinkComponent;
+    if (typeof link === 'string') {
+        linkProps = resolveLinkProps(link, linkType);
+        url = linkProps.to ? linkProps.to : linkProps.href;
+        LinkComponent = linkProps.to ? Link : 'a';
+    }
+
+    let BannerButton;
+    if (showButton !== 'never') {
+        const buttonClass =
+            showButton === 'hover' ? classes.buttonHover : classes.button;
+
+        const handleClick = () => {
+            if (openInNewTab) {
+                window.open(url, '_blank');
+            } else {
+                props.history.push(url);
+            }
+        };
+
+        BannerButton = (
+            <div className={buttonClass}>
+                <Button
+                    priority={typeToPriorityMapping[buttonType]}
+                    type="button"
+                    onClick={handleClick}
+                >
+                    {buttonText}
+                </Button>
+            </div>
+        );
+    }
+
+    const overlayClass =
+        showOverlay === 'hover' && !hovered
+            ? classes.overlayHover
+            : classes.overlay;
+
+    let BannerFragment = (
+        <div className={classes.wrapper} style={wrapperStyles}>
+            <div className={overlayClass} style={overlayStyles}>
+                <div
+                    className={classes.content}
+                    style={contentStyles}
+                    dangerouslySetInnerHTML={toHTML(content)}
+                />
+                {BannerButton}
+            </div>
+        </div>
+    );
+
+    if (typeof link === 'string' && LinkComponent) {
```

Comment:
LinkComponent is only truthy in case where link === 'string' (redundant check).  Perhaps move linkProps/url assignment (129-136) closer to where they're being used here?

Reply:
The link logic is also used in the button, thus needs to be before the button and I don't like the idea of having two conditions to retrieve the component, I've added a comment to help describe this.

Reply:
Ah yeah my bad.  Would be nice to get the respective usages as close to each other as possible.